### PR TITLE
Flannel/CoreDNS/metrics-server update for v1.21

### PR DESCRIPTION
**Pull request by superseb**
#### Pull Request Checklist ####

- [X] Change does not remove any existing Images or Tags in the images-list file
- [X] Change does not remove / overwrite exiting Images or Tags in Rancher DockerHub
- [X] New entries are in format `SOURCE DESTINATION TAG`
- [X] New entries are added to the correct section of the list (sorted lexicographically)
- [X] New entries have a repo created in Rancher Dockerhub (where the image will be mirrored to)
- [X] Changes to scripting or CI config have been tested to the best of your ability

#### Types of Change ####

New images for k8s v1.21 addons updates

#### Linked Issues ####

https://github.com/rancher/rancher/issues/31567#issuecomment-858754265

#### Additional Notes ####

* Missing kube-dns because images for 1.18.0 are missing upstream.
* Calico was already added for 3.19.1

#### Final Checks after the PR is merged ####
- [ ] Confirm that you can pull the new images and tags from DockerHub

### CODE REVIEW

File: images-list [hunk 1]
```diff
@@ -317,6 +319,7 @@ quay.io/cilium/operator-generic rancher/mirrored-cilium-operator-generic v1.9.4
 quay.io/cilium/operator-generic rancher/mirrored-cilium-operator-generic v1.9.6
 quay.io/cilium/operator-generic rancher/mirrored-cilium-operator-generic v1.9.8
 quay.io/cilium/startup-script rancher/mirrored-cilium-startup-script 62bfbe88c17778aad7bef9fa57ff9e2d4a9ba0d8
+quay.io/coreos/flannel rancher/mirrored-coreos-flannel v0.14.0
```
Comment (StrongMonkey) [inline images-list:322]: Do we use upstream flannel? I think `rancher/coreos-flannel:v0.13.0-rancher1` was used in previous releases.

Comment (superseb) [reply images-list:322]: v0.12.0 (previous versions) didnt have manifest support, this one has.

Comment (StrongMonkey) [reply images-list:322]: good :+1:

**Review by StrongMonkey — DISMISSED** [review]

**Review by cbron — APPROVED** [review]
lgtm, will merge in morning

**Pull request merged**